src/group_theory/subgroup/pointwise.lean

```diff
@@ -111,6 +113,14 @@ begin
   exact H.mul_mem hh hh',
 end
 
+/-- The subgroup of powers of an integer of a commutative subgroup.
+
+This is available as an instance in the `pointwise` locale. -/
+protected def pointwise_has_pow {G : Type*} [comm_group G] : has_pow (subgroup G) ℤ :=
```

Comment:
Can you add the nat version too?

Reply:
We don’t have to, because it infers it from the submonoid instance already.

Reply:
Really? How does it do that?

Reply:
I'm not sure, but when I try to define `has_pow (subgroup G) ℕ` it says there is already an instance coming from submonoid.